presto-main/src/main/java/com/facebook/presto/util/FastutilSetHelper.java

```diff
@@ -65,43 +65,156 @@ public static boolean in(boolean booleanValue, BooleanOpenHashSet set)
         return set.contains(booleanValue);
     }
 
-    public static boolean in(byte byteValue, ByteOpenHashSet set)
+    public static boolean in(double doubleValue, DoubleOpenCustomHashSet set)
     {
-        return set.contains(byteValue);
+        return set.contains(doubleValue);
     }
 
-    public static boolean in(char charValue, CharOpenHashSet set)
+    public static boolean in(long longValue, LongOpenCustomHashSet set)
     {
-        return set.contains(charValue);
+        return set.contains(longValue);
     }
 
-    public static boolean in(double doubleValue, DoubleOpenHashSet set)
+    public static boolean in(Object sliceValue, ObjectOpenHashSet<Slice> set)
     {
-        return set.contains(doubleValue);
+        return set.contains(sliceValue);
     }
 
-    public static boolean in(float floatValue, FloatOpenHashSet set)
+    public static boolean in(Object objectValue, ObjectOpenCustomHashSet<?> set)
     {
-        return set.contains(floatValue);
+        return set.contains(objectValue);
     }
 
-    public static boolean in(int intValue, IntOpenHashSet set)
+    private static final class LongStrategy
+            implements LongHash.Strategy
     {
-        return set.contains(intValue);
-    }
+        private MethodHandle hashCodeHandle;
+        private MethodHandle equalsHandle;
 
-    public static boolean in(long longValue, LongOpenHashSet set)
-    {
-        return set.contains(longValue);
+        private LongStrategy()
+        {
+        }
+
+        private LongStrategy(FunctionRegistry registry, Type type)
+        {
+            hashCodeHandle = registry.getScalarFunctionImplementation(registry.resolveOperator(HASH_CODE, ImmutableList.of(type))).getMethodHandle();
+            equalsHandle = registry.getScalarFunctionImplementation(registry.resolveOperator(EQUAL, ImmutableList.of(type, type))).getMethodHandle();
+        }
+
+        @Override
+        public int hashCode(long l)
+        {
+            try {
+                long hash = (long) hashCodeHandle.invokeExact(l);
+                return (int) hash;
+            }
+            catch (Throwable t) {
+                Throwables.propagateIfInstanceOf(t, Error.class);
```

Comment:
You could add a helper that wraps the method handle with the exception handling. `TryCastFunction` does this